chore(travis): code coverage / codecov.io

SkyzohKey opened the pull request:
This pull-request add supports for code coverage. (gcov + codecov.io)

SkyzohKey committed:
chore(travis): call codecov after success

Call CodeCov after build success.

SkyzohKey committed:
chore(travis): update .travis.yml coveralls.

Add $COVERALLS_TOKEN environment var from TravisCI.

SkyzohKey committed:
chore(travis): update .travis.yml coveralls.

Update.

SkyzohKey committed:
chore(travis): update .travis.yml coveralls.

Update.

SkyzohKey committed:
chore(travis): update .travis.yml coveralls.

Update.

SkyzohKey committed:
chore(travis): update .travis.yml coveralls.

Update.

SkyzohKey committed:
chore(travis): add irc notification

Add IRC notifications for #ricin-dev@freenode.net

SkyzohKey committed:
chore(travis): add irc notification

Typo fix.

SkyzohKey committed:
chore(travis): add -Db_coverage=true

Add -Db_coverage=true

SkyzohKey committed:
chore(travis): add --deps to meson

Write make-style dependency information to this file

SkyzohKey committed:
chore(travis): add --enable-gcov

Enable gcov to be used by codecov.

SkyzohKey committed:
chore(travis): add lcov & genhtml dependencies

Mandatory for meson to run coverage tests.

SkyzohKey committed:
chore(travis): revert --enable-gcov until issue fix

This commit revert --enable-gcov flag until https://github.com/mesonbuild/meson/issues/498 is fixed.
Back to `make autogen` for now.

SkyzohKey committed:
chore(travis): try using ninja coverage-html
This should output coverage informations into coverage-html/

GitCop commented:
There were the following issues with your Pull Request

 * Commit: 79f9cac6a1ead32ddbac921ba3fcba765eaf0248
  * Your commit message does not contain a body
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/RicinApp/Ricin/blob/master/CONTRIBUTING.md#-git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

SkyzohKey committed:
chore(travis): update meson to 0.30.0.dev1

Update Meson version from 0.26.0 to 0.30.0.dev1, should fix issues with --enable-gcov

GitCop commented:
There were the following issues with your Pull Request

 * Commit: 79f9cac6a1ead32ddbac921ba3fcba765eaf0248
  * Your commit message does not contain a body
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/RicinApp/Ricin/blob/master/CONTRIBUTING.md#-git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

SkyzohKey committed:
chore(travis): update Meson install command

The old one won't work anymore since we updated.

GitCop commented:
There were the following issues with your Pull Request

 * Commit: 79f9cac6a1ead32ddbac921ba3fcba765eaf0248
  * Your commit message does not contain a body
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/RicinApp/Ricin/blob/master/CONTRIBUTING.md#-git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

SkyzohKey committed:
chore(travis): fix meson path

meson → /usr/bin/local/meson

GitCop commented:
There were the following issues with your Pull Request

 * Commit: 79f9cac6a1ead32ddbac921ba3fcba765eaf0248
  * Your commit message does not contain a body
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/RicinApp/Ricin/blob/master/CONTRIBUTING.md#-git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

SkyzohKey committed:
chore(travis): add -Db_coverage=true

again.

GitCop commented:
There were the following issues with your Pull Request

 * Commit: 79f9cac6a1ead32ddbac921ba3fcba765eaf0248
  * Your commit message does not contain a body
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/RicinApp/Ricin/blob/master/CONTRIBUTING.md#-git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

SkyzohKey committed:
chore(travis): fix meson b_coverage

Add ninja test & ninja coverage html

GitCop commented:
There were the following issues with your Pull Request

 * Commit: 79f9cac6a1ead32ddbac921ba3fcba765eaf0248
  * Your commit message does not contain a body
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/RicinApp/Ricin/blob/master/CONTRIBUTING.md#-git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

SkyzohKey committed:
chore(travis): typo fix

Typo fix: /usr/bin/local → /usr/local/bin

GitCop commented:
There were the following issues with your Pull Request

 * Commit: 79f9cac6a1ead32ddbac921ba3fcba765eaf0248
  * Your commit message does not contain a body
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/RicinApp/Ricin/blob/master/CONTRIBUTING.md#-git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

SkyzohKey committed:
chore(travis): meson now is meson.py

0.30.0.dev1 changed meson executable name from `meson` to `meson.py`

GitCop commented:
There were the following issues with your Pull Request

 * Commit: 79f9cac6a1ead32ddbac921ba3fcba765eaf0248
  * Your commit message does not contain a body
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/RicinApp/Ricin/blob/master/CONTRIBUTING.md#-git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

SkyzohKey committed:
fix(rpm): rpm isn't used on ubuntu (travis)

Since ubuntu doesn't use RPM, builds are failing.

GitCop commented:
There were the following issues with your Pull Request

 * Commit: 79f9cac6a1ead32ddbac921ba3fcba765eaf0248
  * Your commit message does not contain a body
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/RicinApp/Ricin/blob/master/CONTRIBUTING.md#-git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

SkyzohKey committed:
chore(ninja): use `ninja cov`

Use `ninja cov` instead of `ninja -C build test; ninja -C build coverage-html` as stated here: https://github.com/leanprover/lean/blob/master/doc/make/coverage.md

SkyzohKey committed:
chore(travis): typo fix

Forgot to cd into the build folder, without cd'ing it, ninja doesn't find it's ninja.build file. Making the build to fail.

SkyzohKey committed:
chore(ninja): use `ninja -C build cov`

Better than using cd.

SkyzohKey committed:
chore(travis): use `ninja coverage-xml`

Try to generate code coverage in XML format.

SkyzohKey committed:
chore(gcovr): compile Gcovr from source

Compile Gcovr from sources, here's the backport for Ubuntu Precise 12.04: https://github.com/eraserix/gcovr

SkyzohKey committed:
chore(gcovr): typo fix

cd 3.1 → cd gcovr-3.1

SkyzohKey committed:
chore(travis): add Python2.7 for gcovr to compile
Gcovr needs Python2.7 in order to compile,
this commit adds it to the dependencies added.

GitCop commented:
There were the following issues with your Pull Request

 * Commit: 7a5d3d5f8861349b48c7f012cd5df729e4a75e99
  * Your commit message does not contain a body

Guidelines are available at https://github.com/RicinApp/Ricin/blob/master/CONTRIBUTING.md#-git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

SkyzohKey committed:
chore(travis): typo fix

python3 → python

SkyzohKey committed:
chore(meson): build from latest trunk

Don't use releases to build Meson, this avoid compatibility issues.

SkyzohKey committed:
chore(ninja): move coverage reports after success

Run coverage reports generation only if build succeed.

SkyzohKey committed:
chore(meson): fix path

Fixed meson path.

SkyzohKey committed:
chore(travis): typo fix

`mkdir -p` → `mkdir -p build`

SkyzohKey committed:
chore(travis): sudoed ninja + ldconfig

`ninja coverage-html` needs sudo, else it fails with "Permission denied".
Also, added a `ldconfig` line for libconfig, else Ricin doesn't find it.

SkyzohKey committed:
chore(travis): sudoed ninja build & ninja test

`ninja -C build` and `ninja -C build test` needed sudo.

SkyzohKey committed:
Merge branch 'master' into chore/travis-coverage

SkyzohKey committed:
fix(settings): move settings init after files check

Moved the init Settings line under files checks in order to avoid settings trying to open unexisting files.

SkyzohKey committed:
Merge branch 'master' into chore/travis-coverage

SkyzohKey committed:
chore(travis): enable ccache

Add ccache to have built-in dependencies caching, in order to speed builds time.

SkyzohKey committed:
chore(travis): add clang compiler to compilers list

This commit is a test for the clang compiler.

Commit:
fix(conflicts): removed rpm spec generator

Commit:
Merge branch 'chore/travis-coverage' of github.com:RicinApp/Ricin into chore/travis-coverage

SkyzohKey committed:
Merge branch 'master' into chore/travis-coverage

SkyzohKey committed:
Merge branch 'master' into chore/travis-coverage

SkyzohKey committed:
Merge branch 'master' into chore/travis-coverage

SkyzohKey committed:
Merge branch 'master' into chore/travis-coverage
